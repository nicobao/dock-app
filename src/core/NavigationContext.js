import React, {useEffect, useState} from 'react';

export const NavigationContext = React.createContext({currentTab: 'tokens'});
export const withNavigationContext = WrappedComponent => props => {
  const [currentTab, setCurrentTab] = useState('tokens');
  const {route} = props;

  useEffect(() => {
    if (route && route?.params?.currentTab) {
      setCurrentTab(route.params.currentTab);
    }
  }, [route]);

  return (
    <NavigationContext.Provider
      value={{
        currentTab,
      }}>
      <WrappedComponent {...props} />
    </NavigationContext.Provider>
  );
};
