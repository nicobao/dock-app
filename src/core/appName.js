export const app = {
  name: 'Dock Wallet',
};
