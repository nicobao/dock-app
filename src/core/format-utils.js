import {
  formatCurrency,
  formatAddress,
  formatDockAmount,
  getPlainDockAmount,
  formatDate,
} from '@docknetwork/wallet-sdk-core/lib/core/format-utils';

export {
  formatCurrency,
  formatAddress,
  formatDockAmount,
  getPlainDockAmount,
  formatDate,
};
