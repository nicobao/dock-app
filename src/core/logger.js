const winston = require('winston');

export const Logger = winston.createLogger({
  format: winston.format.simple(),
  level: 'info',
  transports: [new winston.transports.Console()],
});
