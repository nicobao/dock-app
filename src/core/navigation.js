import {createNavigationContainerRef} from '@react-navigation/native';

export const navigationRef = createNavigationContainerRef();

export function navigate(name, params) {
  if (navigationRef.isReady()) {
    navigationRef.navigate(name, params);
  }
}

export function navigateBack() {
  if (navigationRef.isReady()) {
    navigationRef.goBack();
  }
}
