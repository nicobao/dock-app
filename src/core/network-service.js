import axios from 'axios';
import {RequestLogger} from '@docknetwork/wallet-sdk-request-logger/lib/request-logger';
const instance = axios.create({});
import store from '../core/redux-store';
instance.interceptors.response.use(
  function (response) {
    const {url, method, headers, data} = response.config;
    const {status, data: payload} = response;

    const shouldLogRequest =
      store.getState()?.app?.features?.shouldLogRequest || false;
    if (shouldLogRequest) {
      RequestLogger.logRequest({
        url,
        method,
        headers,
        body: data,
        response: payload,
        status,
      });
    }
    return response;
  },
  function (error) {
    const shouldLogRequest =
      store.getState()?.app?.features?.shouldLogRequest || false;
    const response = error.response;
    const {status, data: payload} = response;

    const {url, method, headers, data} = response.config;
    if (shouldLogRequest) {
      RequestLogger.logRequest({
        url,
        method,
        headers,
        body: data,
        response: payload,
        status,
      });
    }
    return Promise.reject(error);
  },
);

export default instance;
