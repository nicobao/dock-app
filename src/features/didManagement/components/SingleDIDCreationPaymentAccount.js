import {NBox, Typography} from '../../../design-system';
import React from 'react';
import {HStack, Stack} from 'native-base';
import {TokenAmount} from '../../tokens/ConfirmTransactionModal';
export function SingleDIDCreationPaymentAccount({item}) {
  return (
    <NBox>
      <HStack mb={1}>
        <Typography
          style={{
            flexGrow: 1,
          }}
          numberOfLines={1}
          textAlign="left"
          variant="description">
          {item.label}
        </Typography>
        <TokenAmount amount={item.balance}>
          {({tokenAmount, tokenSymbol}) => (
            <>
              <Stack direction="column">
                <Typography variant="description">
                  {tokenAmount} {tokenSymbol}
                </Typography>
              </Stack>
            </>
          )}
        </TokenAmount>
      </HStack>
      <Typography
        numberOfLines={1}
        textAlign="left"
        variant="screen-description">
        {item.description}
      </Typography>
    </NBox>
  );
}
