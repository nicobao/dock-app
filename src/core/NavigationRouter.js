import {NavigationContainer} from '@react-navigation/native';

import React, {useEffect} from 'react';
import {Linking, Platform} from 'react-native';

import {navigationRef} from './navigation';

import DeepLinking from 'react-native-deep-linking';

import {authenticationSelectors} from '../features/unlock-wallet/unlock-wallet-slice';
import {useSelector} from 'react-redux';
import {useDeepLink} from './hooks/deepLinkHooks';
import {UnAuthNavigationStackScreen} from './navigation/unauthenticated-stack';
import {AuthNavigationStackScreen} from './navigation/authenticated-stack';

const forFade = ({current, closing}) => ({
  cardStyle: {
    opacity: current.progress,
  },
});

const screenOptions = {
  headerShown: false,
};

if (Platform.OS === 'android') {
  screenOptions.cardStyleInterpolator = forFade;
}

export function NavigationRouter() {
  const isLoggedIn = useSelector(authenticationSelectors.isLoggedIn);
  const {gotoScreenDeepLink} = useDeepLink({isLoggedIn});

  useEffect(() => {
    const getAsyncURL = async () => {
      const initialUrl = await Linking.getInitialURL();
      if (initialUrl !== undefined && initialUrl != null) {
        gotoScreenDeepLink({url: initialUrl});
      }
    };

    getAsyncURL();
  }, [gotoScreenDeepLink]);

  useEffect(() => {
    DeepLinking.addScheme('dockwallet://');
    Linking.addEventListener('url', gotoScreenDeepLink);

    return () => {
      Linking.removeEventListener('url', gotoScreenDeepLink);
    };
  }, [gotoScreenDeepLink]);

  return (
    <NavigationContainer ref={navigationRef}>
      {isLoggedIn ? (
        <AuthNavigationStackScreen />
      ) : (
        <UnAuthNavigationStackScreen />
      )}
    </NavigationContainer>
  );
}
