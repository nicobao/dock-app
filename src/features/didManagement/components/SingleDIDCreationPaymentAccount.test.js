import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {SingleDIDCreationPaymentAccount} from './SingleDIDCreationPaymentAccount';
import {renderAppProviders} from '../../../core/test-utils';

const mockStore = configureMockStore();

describe('SingleDIDCreationPaymentAccount', () => {
  it('should render correctly ', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      renderAppProviders(
        <SingleDIDCreationPaymentAccount
          item={{
            value: '3C7Hq5jQGxeYzL7LnVASn48tEfr6D7yKtNYSuXcgioQoWWsB',
            label: 'Test Account',
            description: '3C7Hq5jQGxeYzL7LnVASn48tEfr6D7yKtNYSuXcgioQoWWsB',
            balance: '0.01',
            fetchBalance: jest.fn(),
          }}
        />,
      ),
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
