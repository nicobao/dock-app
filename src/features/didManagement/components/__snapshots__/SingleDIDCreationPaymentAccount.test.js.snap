// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleDIDCreationPaymentAccount should render correctly  1`] = `
<NativeBaseProvider
  initialWindowMetrics={
    Object {
      "frame": Object {
        "height": 0,
        "width": 0,
        "x": 0,
        "y": 0,
      },
      "insets": Object {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      },
    }
  }
>
  <SingleDIDCreationPaymentAccount
    item={
      Object {
        "balance": "0.01",
        "description": "3C7Hq5jQGxeYzL7LnVASn48tEfr6D7yKtNYSuXcgioQoWWsB",
        "fetchBalance": [MockFunction],
        "label": "Test Account",
        "value": "3C7Hq5jQGxeYzL7LnVASn48tEfr6D7yKtNYSuXcgioQoWWsB",
      }
    }
  />
</NativeBaseProvider>
`;
